matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/space/DefaultSpaceService.kt

```diff
@@ -105,8 +108,10 @@ internal class DefaultSpaceService @Inject constructor(
         return roomSummaryDataSource.getSpaceSummaries(spaceSummaryQueryParams, sortOrder)
     }
 
-    override fun getRootSpaceSummaries(): List<RoomSummary> {
-        return roomSummaryDataSource.getRootSpaceSummaries()
+    override suspend fun getRootSpaceSummaries(): List<RoomSummary> {
+        return withContext(coroutineDispatchers.io) {
```

Comment:
:+1: 

not for this PR as it would introduce lots of changes, but worth highlighting in case we want guard against other cases

from my understanding the general approach of coroutines scoping is to define the context switching at the lowest possible level, which in this case would be 

```
monarchy.fetchAllMappedSync
```
 
we already have a similar wrapper for [awaitTransaction](https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/AsyncTransaction.kt#L38)

this would also force non coroutine based usages to rely on `runBlocking`, which would have the benefit of make finding the usages a bit easier

Reply:
I agree, good point!

Reply:
Actually we might want to dispatch querying db to a background thread, to avoid blocking the dispatcher, as Retrofit does for request (and probably Room)